asset-enrichment/src/main/scala/org/sunbird/job/functions/AssetEnrichmentEventRouter.scala

```diff
@@ -0,0 +1,47 @@
+package org.sunbird.job.functions
+
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.slf4j.LoggerFactory
+import org.sunbird.job.domain.Event
+import org.sunbird.job.task.AssetEnrichmentConfig
+import org.sunbird.job.{BaseProcessFunction, Metrics}
+
+class AssetEnrichmentEventRouter(config: AssetEnrichmentConfig)
+  extends BaseProcessFunction[Event, String](config) {
+
+  private[this] val logger = LoggerFactory.getLogger(classOf[AssetEnrichmentEventRouter])
+
+  override def open(parameters: Configuration): Unit = {
+    super.open(parameters)
+  }
+
+  override def close(): Unit = {
+    super.close()
+  }
+
+  override def processElement(event: Event, context: ProcessFunction[Event, String]#Context, metrics: Metrics): Unit = {
+    logger.info(s"Processing event for AssetEnrichment for identifier : ${event.id}")
+    val message = event.validate(config.maxIterationCount)
+    if (message.isEmpty) {
+      metrics.incCounter(config.totalEventsCount)
```

Comment:
```suggestion
      metrics.incCounter(config.totalEventsCount)
```
this should be outside the if condition.